wasmbinding/stargate_query_test.go

```diff
@@ -0,0 +1,143 @@
+		// bank
+		"/cosmos.bank.v1beta1.Query/SupplyOf",
+		"/cosmos.bank.v1beta1.Query/Params",
+		"/cosmos.bank.v1beta1.Query/DenomMetadata",
+		"/cosmos.bank.v1beta1.Query/AllBalances",
+		"/cosmos.bank.v1beta1.Query/Balance",
+		// gov
+		"/cosmos.gov.v1.Query/Proposal",
+		"/cosmos.gov.v1.Query/Params",
+		"/cosmos.gov.v1.Query/Vote",
+
+		// ibc
+		"/ibc.core.client.v1.Query/ClientState",
+		"/ibc.core.client.v1.Query/ConsensusState",
+		"/ibc.core.connection.v1.Query/Connection",
+		"/ibc.core.connection.v1.Query/Connections",
+		"/ibc.core.connection.v1.Query/ClientConnections",
+		"/ibc.core.connection.v1.Query/ConnectionConsensusState",
+		"/ibc.core.connection.v1.Query/ConnectionParams",
+
+		// ibc transfer
+		"/ibc.applications.transfer.v1.Query/DenomTrace",
+		"/ibc.applications.transfer.v1.Query/Params",
+		"/ibc.applications.transfer.v1.Query/DenomHash",
+		"/ibc.applications.transfer.v1.Query/EscrowAddress",
+		"/ibc.applications.transfer.v1.Query/TotalEscrowForDenom",
+	} {
+		stargateQueryPaths.Add(queryPath)
+	}
+
+	t.Log("stargateQueryPaths: Add nibiru query paths from GRPC service descriptions")
+	queryServiceDescriptions := []grpc.ServiceDesc{
+		epochs.GrpcQueryServiceDesc(),
+		devgas.GrpcQueryServiceDesc(),
+		inflation.GrpcQueryServiceDesc(),
+		oracle.GrpcQueryServiceDesc(),
+		sudotypes.GrpcQueryServiceDesc(),
+		tokenfactory.GrpcQueryServiceDesc(),
+	}
+	for _, serviceDesc := range queryServiceDescriptions {
+		for _, queryMethod := range serviceDesc.Methods {
+			stargateQueryPaths.Add(
+				fmt.Sprintf("/%v/%v", serviceDesc.ServiceName, queryMethod.MethodName),
+			)
+		}
+	}
+
+	gotQueryPaths := []string{}
+	// It's not required for the response type and the method description of the
+	// stargate query's gRPC path to match up exactly as expected. The exception
+	// to this convention is when our response type doesn't stripped of its
+	// "Response" suffix and "Query" prefix is not the same as the method name.
+	// This happens when "QueryAAARequest" does not return a "QueryAAAResponse".
+	exceptionPaths := set.New[string]("/nibiru.oracle.v1.QueryExchangeRateResponse")
+	for queryPath, protobufResponse := range wasmbinding.WasmAcceptedStargateQueries() {
+		gotQueryPaths = append(gotQueryPaths, queryPath)
+
+		// Show that the underlying protobuf name and query paths coincide.
+		pbQueryResponseTypeUrl := "/" + proto.MessageName(protobufResponse)
+		isExceptionPath := exceptionPaths.Has(pbQueryResponseTypeUrl)
+		splitResponse := strings.Split(pbQueryResponseTypeUrl, "Response")
+		assert.Lenf(t, splitResponse, 2, "typeUrl: %v",
+			splitResponse, pbQueryResponseTypeUrl)
+
+		// Get proto message "package" from the response type
+		typeUrlMinusSuffix := splitResponse[0]
+		typeUrlPartsFromProtoMsg := strings.Split(typeUrlMinusSuffix, ".")
+		assert.GreaterOrEqual(t, len(typeUrlPartsFromProtoMsg), 4, typeUrlPartsFromProtoMsg)
+		protoMessagePackage := typeUrlPartsFromProtoMsg[:3]
```

Comment:
I'm not sure if 3 is the right amount in all cases. For example, the IBC core ones are `ibc.core.connection.v1`.

Reply:
I think you're right. For on where we assert >= instead of >, the package would be all except that last element rather than the first 3. 

Addressed in a69b501ed12e17c48e58e254e62efae239956588